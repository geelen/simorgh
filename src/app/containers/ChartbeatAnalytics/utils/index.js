import Cookie from 'js-cookie';
import onClient from '../../../lib/utilities/onClient';
import { getPromoHeadline } from '../../../lib/analyticsUtils/article';
import { getPageTitle } from '../../../lib/analyticsUtils/frontpage';
import { getReferrer } from '../../../lib/analyticsUtils';

const ID_COOKIE = 'ckns_sylphid';

export const chartbeatUID = 50924;
export const useCanonical = true;
export const chartbeatSource = '//static.chartbeat.com/js/chartbeat.js';

const buildSectionArr = (service, value, type) => [
  `${service} - ${value}`,
  ...(type ? [`${service} - ${value} - ${type}`] : []),
];

const buildServiceType = (service, type) => [`${service} - ${type}`];

const capitalize = s => s.charAt(0).toUpperCase() + s.slice(1);

export const getSylphidCookie = () =>
  onClient() ? Cookie.get(ID_COOKIE) : null;

export const getDomain = service => {
  const serviceLower = service.toLowerCase();

  return serviceLower === 'news' ? 'bbc.co.uk' : `${service}.bbc.co.uk`;
};

export const getType = (pageType, shorthand = false) => {
  switch (pageType) {
    case 'frontPage':
    case 'index':
      return shorthand ? 'IDX' : 'Index';
    case 'article':
      return shorthand ? 'ART' : 'New Article';
    default:
      return null;
  }
};

export const buildSections = (service, type, producer, chapter) => {
  const addProducer = producer && service !== producer;
  const serviceCap = capitalize(service);
  const pageType = getType(type, true);

  const parts = [
    serviceCap,
    ...(type ? buildServiceType(serviceCap, pageType) : []),
    ...(addProducer ? buildSectionArr(serviceCap, producer, pageType) : []),
    ...(chapter ? buildSectionArr(serviceCap, chapter, pageType) : []),
  ];

  return parts.join(', ');
};

export const getTitle = (pageType, pageData, brandName) => {
  switch (pageType) {
    case 'frontPage':
    case 'index':
      return getPageTitle(pageData, brandName);
    case 'article':
      return getPromoHeadline(pageData);
    default:
      return null;
  }
};

export const getConfig = ({
  isAmp,
  platform,
  pageType,
  data,
  brandName,
  env,
  service,
  origin,
  previousPath,
}) => {
  const referrer = getReferrer(platform, origin, previousPath);
  const title = getTitle(pageType, data, brandName);
  const domain = env !== 'live' ? getDomain('test') : getDomain(service);
  const sections = buildSections(service, pageType);
  const cookie = getSylphidCookie();
  const type = getType(pageType);
  const currentPath = onClient() && window.location.pathname;
  return {
    domain,
    sections,
    uid: chartbeatUID,
    title,
    virtualReferrer: referrer,
    ...(isAmp && { contentType: type }),
    ...(!isAmp && {
      type,
      useCanonical,
      path: currentPath,
    }),
    ...(cookie && { idSync: { bbc_hid: cookie } }),
  };
};
