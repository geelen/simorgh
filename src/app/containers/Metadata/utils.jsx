import React from 'react';

export const getIconAssetUrl = (service, size) =>
  `https://news.files.bbci.co.uk/include/articles/public/${service}/images/icons/icon-${size}.png`;

const createIconLinks = (service, iconSizes, iconType) => {
  return iconSizes.map(size => {
    const iconAssetUrl = getIconAssetUrl(service, size);
    const key = `${service}-${size}`;

    if (iconType === 'icon') {
      return (
        <link
          key={key}
          rel="icon"
          type="image/png"
          href={iconAssetUrl}
          sizes={size}
        />
      );
    }

    return (
      <link key={key} rel="apple-touch-icon" sizes={size} href={iconAssetUrl} />
    );
  });
};

export const getIconLinks = (service, iconSizes) => {
  if (!iconSizes) {
    return null;
  }
  const iconTypes = Object.keys(iconSizes);
  return iconTypes.map(iconType => {
    return createIconLinks(service, iconSizes[iconType], iconType);
  });
};

export const renderAmpHtml = (ampLink, isAmp) => {
  if (isAmp) {
    return null;
  }
  return <link rel="amphtml" href={ampLink} />;
};

export const getAppleTouchUrl = service => {
  const assetsPath = process.env.SIMORGH_PUBLIC_STATIC_ASSETS_PATH || '/';
  const separatorSlash = assetsPath[assetsPath.length - 1] !== '/' ? '/' : '';

  return [
    process.env.SIMORGH_PUBLIC_STATIC_ASSETS_ORIGIN,
    assetsPath,
    separatorSlash,
    service,
    '/images/icons/icon-192x192.png',
  ].join('');
};

export const renderAlternateLinks = link => (
  <link
    rel="alternate"
    href={link.href}
    hrefLang={link.hrefLang}
    key={link.hrefLang}
  />
);
