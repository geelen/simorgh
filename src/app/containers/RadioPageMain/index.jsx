import React, { useContext } from 'react';
import { string, shape, object, arrayOf } from 'prop-types';
import path from 'ramda/src/path';
import ATIAnalytics from '../ATIAnalytics';
import MetadataContainer from '../Metadata';
import Grid, { GelPageGrid } from '#app/components/Grid';
import LinkedData from '../LinkedData';
import RadioPageBlocks from '../RadioPageBlocks';
import { ServiceContext } from '../../contexts/ServiceContext';

const RadioPageMain = ({ pageData }) => {
  const blocks = path(['content', 'blocks'], pageData);
  const promo = path(['promo'], pageData);
  const metadata = path(['metadata'], pageData);
  const { dir } = useContext(ServiceContext);
  return (
    <>
      <ATIAnalytics data={pageData} />
      <MetadataContainer
        title={promo.name}
        lang={metadata.language}
        description={promo.summary}
        openGraphType="website"
      />
      <LinkedData type="RadioChannel" seoTitle={promo.name} />

      <GelPageGrid
        forwardedAs="main"
        role="main"
        dir={dir}
        columns={{
          group0: 6,
          group1: 6,
          group2: 6,
          group3: 6,
          group4: 8,
          group5: 20,
        }}
        enableGelGutters
      >
        <Grid
          item
          dir={dir}
          startOffset={{
            group0: 1,
            group1: 1,
            group2: 1,
            group3: 1,
            group4: 2,
            group5: 5,
          }}
          columns={{
            group0: 6,
            group1: 6,
            group2: 6,
            group3: 6,
            group4: 6,
            group5: 12,
          }}
          margins={{ group0: true, group1: true, group2: true, group3: true }}
        >
          <RadioPageBlocks blocks={blocks} />
        </Grid>
      </GelPageGrid>
    </>
  );
};

RadioPageMain.propTypes = {
  pageData: shape({
    metadata: shape({
      id: string,
      tags: object,
    }),
    promo: shape({
      subtype: string,
      name: string,
    }),
    content: shape({
      blocks: arrayOf(
        shape({
          uuid: string,
          id: string,
          externalId: string,
          text: string,
          type: string,
        }),
      ),
    }),
  }).isRequired,
};

export default RadioPageMain;
