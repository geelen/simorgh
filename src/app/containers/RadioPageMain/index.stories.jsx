import React from 'react';
import { storiesOf } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import { withServicesKnob } from '@bbc/psammead-storybook-helpers';
import { withKnobs } from '@storybook/addon-knobs';
import WithTimeMachine from '#testHelpers/withTimeMachine';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import RadioPageMain from '.';
import indonesia from '#data/indonesia/bbc_indonesian_radio/liveradio.json';
import korean from '#data/korean/bbc_korean_radio/liveradio.json';
import tigrinya from '#data/tigrinya/bbc_tigrinya_radio/liveradio.json';
import afaanoromoo from '#data/afaanoromoo/bbc_afaanoromoo_radio/liveradio.json';
import amharic from '#data/amharic/bbc_amharic_radio/liveradio.json';

// Not all services have fixtures for article data yet
// the service selector will be constrained to services that have article fixtures:
const liveRadioFixtures = {
  indonesia,
  korean,
  tigrinya,
  afaanoromoo,
  amharic,
};

const validServices = Object.keys(liveRadioFixtures);

const matchFixtures = service => ({
  params: {
    mediaId: 'liveradio',
    serviceId: {
      indonesia: 'bbc_indonesian_radio',
      korean: 'bbc_korean_radio',
      tigrinya: 'bbc_tigrinya_radio',
      afaanoromoo: 'bbc_afaanoromoo_radio',
      amharic: 'bbc_amharic_radio',
    }[service],
  },
});

storiesOf('Main|Radio Page', module)
  .addDecorator(story => <WithTimeMachine>{story()}</WithTimeMachine>)
  .addDecorator(withKnobs)
  .addDecorator(
    withServicesKnob({
      defaultService: 'indonesia',
      services: validServices,
    }),
  )
  .add('default', ({ service }) => {
    return (
      <ToggleContextProvider>
        <ServiceContextProvider service={service}>
          <RequestContextProvider
            isAmp={false}
            pageType="media"
            origin="https://www.bbc.com"
            service={service}
          >
            <BrowserRouter>
              <RadioPageMain
                pageData={liveRadioFixtures[service]}
                match={matchFixtures(service)}
                service={service}
              />
            </BrowserRouter>
          </RequestContextProvider>
        </ServiceContextProvider>
      </ToggleContextProvider>
    );
  });
