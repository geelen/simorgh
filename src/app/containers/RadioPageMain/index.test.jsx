import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { matchSnapshotAsync } from '@bbc/psammead-test-helpers';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import RadioPageMain from '.';
import amharicPageData from '#data/amharic/bbc_amharic_radio/liveradio';
import preprocessor from '#lib/utilities/preprocessor';
import * as analyticsUtils from '#lib/analyticsUtils';
import { radioPagePreprocessorRules } from '#app/routes/getInitialData/utils/preprocessorRulesConfig';

analyticsUtils.getAtUserId = jest.fn();

describe('Radio Page Main', () => {
  it('should match snapshot for Canonical', async () => {
    const pageData = await preprocessor(
      amharicPageData,
      radioPagePreprocessorRules,
    );

    await matchSnapshotAsync(
      <ServiceContextProvider service="amharic">
        <RequestContextProvider
          bbcOrigin="https://www.test.bbc.co.uk"
          isAmp={false}
          pageType="media"
          pathname="/pathname"
          service="amharic"
          statusCode={200}
        >
          <BrowserRouter>
            <RadioPageMain service="amharic" pageData={pageData} />
          </BrowserRouter>
        </RequestContextProvider>
      </ServiceContextProvider>,
    );
  });

  it('should match snapshot for AMP', async () => {
    const pageData = await preprocessor(
      amharicPageData,
      radioPagePreprocessorRules,
    );

    await matchSnapshotAsync(
      <ServiceContextProvider service="amharic">
        <RequestContextProvider
          bbcOrigin="https://www.test.bbc.co.uk"
          isAmp
          pageType="media"
          pathname="/pathname"
          service="amharic"
          statusCode={200}
        >
          <BrowserRouter>
            <RadioPageMain service="amharic" pageData={pageData} />
          </BrowserRouter>
        </RequestContextProvider>
      </ServiceContextProvider>,
    );
  });
});
