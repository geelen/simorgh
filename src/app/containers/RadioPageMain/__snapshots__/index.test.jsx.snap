// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Page Main should match snapshot for AMP 1`] = `
<html
  amp=""
  dir="ltr"
  lang="am"
>
  <head>
    <title>
      ያድምጡ - BBC News አማርኛ
    </title>
    <link
      href="https://www.test.bbc.com/pathname"
      rel="canonical"
    />
    <link
      href="http://localhost:7080/amharic/images/icons/icon-192x192.png"
      rel="apple-touch-icon"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-72x72.png"
      rel="apple-touch-icon"
      sizes="72x72"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-96x96.png"
      rel="apple-touch-icon"
      sizes="96x96"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-128x128.png"
      rel="apple-touch-icon"
      sizes="128x128"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-144x144.png"
      rel="apple-touch-icon"
      sizes="144x144"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-152x152.png"
      rel="apple-touch-icon"
      sizes="152x152"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-192x192.png"
      rel="apple-touch-icon"
      sizes="192x192"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-384x384.png"
      rel="apple-touch-icon"
      sizes="384x384"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-512x512.png"
      rel="apple-touch-icon"
      sizes="512x512"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-72x72.png"
      rel="icon"
      sizes="72x72"
      type="image/png"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-96x96.png"
      rel="icon"
      sizes="96x96"
      type="image/png"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-192x192.png"
      rel="icon"
      sizes="192x192"
      type="image/png"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-512x512.png"
      rel="apple-touch-startup-image"
    />
    <link
      href="/favicon.ico"
      rel="shortcut icon"
      type="image/x-icon"
    />
    <meta
      content="IE=edge"
      http-equiv="X-UA-Compatible"
    />
    <meta
      charset="utf-8"
    />
    <meta
      content="noodp,noydir"
      name="robots"
    />
    <meta
      content="#B80000"
      name="theme-color"
    />
    <meta
      content="width=device-width, initial-scale=1, minimum-scale=1"
      name="viewport"
    />
    <meta
      content="BBC News አማርኛ"
      name="apple-mobile-web-app-title"
    />
    <meta
      content="BBC News አማርኛ"
      name="application-name"
    />
    <meta
      content="ዝግጅቶቻችንን’"
      name="description"
    />
    <meta
      content="100004154058350"
      name="fb:admins"
    />
    <meta
      content="1609039196070050"
      name="fb:app_id"
    />
    <meta
      content="yes"
      name="mobile-web-app-capable"
    />
    <meta
      content="#B80000"
      name="msapplication-TileColor"
    />
    <meta
      content="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-144x144.png"
      name="msapplication-TileImage"
    />
    <meta
      content="ዝግጅቶቻችንን’"
      name="og:description"
    />
    <meta
      content="https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png"
      name="og:image"
    />
    <meta
      content="BBC News አማርኛ"
      name="og:image:alt"
    />
    <meta
      content="am-ET"
      name="og:locale"
    />
    <meta
      content="BBC News አማርኛ"
      name="og:site_name"
    />
    <meta
      content="ያድምጡ - BBC News አማርኛ"
      name="og:title"
    />
    <meta
      content="website"
      name="og:type"
    />
    <meta
      content="https://www.test.bbc.com/pathname"
      name="og:url"
    />
    <meta
      content="summary_large_image"
      name="twitter:card"
    />
    <meta
      content="@bbcnews"
      name="twitter:creator"
    />
    <meta
      content="ዝግጅቶቻችንን’"
      name="twitter:description"
    />
    <meta
      content="BBC News አማርኛ"
      name="twitter:image:alt"
    />
    <meta
      content="https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png"
      name="twitter:image:src"
    />
    <meta
      content="@bbcnews"
      name="twitter:site"
    />
    <meta
      content="ያድምጡ - BBC News አማርኛ"
      name="twitter:title"
    />
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"RadioChannel","url":"https://www.test.bbc.com/pathname","publisher":{"@type":"NewsMediaOrganization","name":"BBC News አማርኛ","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png"},"thumbnailUrl":"https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.test.bbc.com/pathname","name":"ያድምጡ"}}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c3 {
  font-size: 1.75rem;
  line-height: 2rem;
  font-family: "Noto Sans Ethiopic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #3F3F42;
  display: block;
  margin: 0;
  padding: 2rem 0;
}

.c7 {
  height: 165px;
  position: relative;
  overflow: hidden;
}

.c6 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 50rem;
  max-width: calc(100vw - 2rem);
}

.c4 {
  font-size: 1rem;
  line-height: 1.375rem;
  font-family: "Noto Sans Ethiopic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c2 {
    padding: 0 0.5rem;
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c2 {
    padding: 0 0.5rem;
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c2 {
    padding: 0 1rem;
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c2 {
    padding: 0 1rem;
    width: calc(6/6*(100% - 6 * 1rem) + 5 * 1rem - 0%);
    margin: 0 0.5rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c2 {
    width: calc(6/8*(100% - 8 * 1rem) + 5 * 1rem );
    margin: 0 0.5rem;
    margin-left: 12.5%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c2 {
    width: calc(12/20*(100% - 20 * 1rem) + 11 * 1rem );
    margin: 0 0.5rem;
    margin-left: 20%;
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c2 {
    display: block;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    margin: 0 auto;
    max-width: 63rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    margin: 0 auto;
    max-width: 80rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2rem;
    line-height: 2.625rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 2.75rem;
    line-height: 3.625rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    padding: 2.5rem 0;
  }
}

@media (min-width:63rem) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.75rem;
  }
}

<div>
      <amp-analytics>
        <script
          type="application/json"
        >
          {"transport":{"beacon":false,"xhrpost":false,"image":true},"requests":{"pageview":"\${base}s=598343&s2=4&p=amharic.bbc_amharic_radio.liveradio.page&r=\${screenWidth}x\${screenHeight}x\${screenColorDepth}&re=\${availableScreenWidth}x\${availableScreenHeight}&hl=\${timestamp}&lng=\${browserLanguage}&x1=[urn:bbc:ares::ws_live:bbc_amharic_radio]&x2=[amp]&x3=[news-amharic]&x4=[am]&x5=[\${sourceUrl}]&x6=[\${documentReferrer}]&x7=[player-live]&x8=[simorgh]&x9=[ያድምጡ+-+BBC+News+አማርኛ]"},"triggers":{"trackPageview":{"on":"visible","request":"pageview"}}}
        </script>
      </amp-analytics>
      <main
        class="c0 c1"
        dir="ltr"
        role="main"
      >
        <div
          class="c2"
          dir="ltr"
        >
          <h1
            class="c3"
            id="content"
            tabindex="-1"
          >
            ያድምጡ
          </h1>
          <p
            class="c4"
          >
            ዝግጅቶቻችንን’
          </p>
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <amp-iframe
                  allowfullscreen="allowfullscreen"
                  frameborder="0"
                  layout="fill"
                  sandbox="allow-scripts allow-same-origin"
                  scrolling="no"
                  src="https://polling.test.bbc.co.uk/ws/av-embeds/media/bbc_amharic_radio/liveradio/am/amp"
                  title="Audio player"
                >
                  <amp-img
                    alt=""
                    attribution=""
                    height="16"
                    layout="fill"
                    placeholder="true"
                    src="http://localhost:7080/images/amp_audio_placeholder.png"
                    width="9"
                  />
                </amp-iframe>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>
</html>
`;

exports[`Radio Page Main should match snapshot for Canonical 1`] = `
.c3 {
  font-size: 1.75rem;
  line-height: 2rem;
  font-family: "Noto Sans Ethiopic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  color: #3F3F42;
  display: block;
  margin: 0;
  padding: 2rem 0;
}

.c7 {
  height: 165px;
  position: relative;
  overflow: hidden;
}

.c6 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 50rem;
  max-width: calc(100vw - 2rem);
}

.c4 {
  font-size: 1rem;
  line-height: 1.375rem;
  font-family: "Noto Sans Ethiopic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #3F3F42;
  padding-bottom: 1.5rem;
  margin: 0;
}

.c8 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 0;
  overflow: hidden;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c2 {
    padding: 0 0.5rem;
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c2 {
    padding: 0 0.5rem;
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c2 {
    padding: 0 1rem;
    width: calc(6/6*(100% - 6 * 0.5rem) + 5 * 0.5rem - 0%);
    margin: 0 0.25rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c2 {
    padding: 0 1rem;
    width: calc(6/6*(100% - 6 * 1rem) + 5 * 1rem - 0%);
    margin: 0 0.5rem;
    margin-left: 0%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c2 {
    width: calc(6/8*(100% - 8 * 1rem) + 5 * 1rem );
    margin: 0 0.5rem;
    margin-left: 12.5%;
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c2 {
    width: calc(12/20*(100% - 20 * 1rem) + 11 * 1rem );
    margin: 0 0.5rem;
    margin-left: 20%;
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c2 {
    display: block;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    margin: 0 auto;
    max-width: 63rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    margin: 0 auto;
    max-width: 80rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 2rem;
    line-height: 2.625rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 2.75rem;
    line-height: 3.625rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    padding: 2.5rem 0;
  }
}

@media (min-width:63rem) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1rem;
    line-height: 1.75rem;
  }
}

<html
  dir="ltr"
  lang="am"
>
  <head>
    <title>
      ያድምጡ - BBC News አማርኛ
    </title>
    <link
      href="https://www.test.bbc.com/pathname"
      rel="canonical"
    />
    <link
      href="https://www.test.bbc.co.uk/pathname.amp"
      rel="amphtml"
    />
    <link
      href="http://localhost:7080/amharic/images/icons/icon-192x192.png"
      rel="apple-touch-icon"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-72x72.png"
      rel="apple-touch-icon"
      sizes="72x72"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-96x96.png"
      rel="apple-touch-icon"
      sizes="96x96"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-128x128.png"
      rel="apple-touch-icon"
      sizes="128x128"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-144x144.png"
      rel="apple-touch-icon"
      sizes="144x144"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-152x152.png"
      rel="apple-touch-icon"
      sizes="152x152"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-192x192.png"
      rel="apple-touch-icon"
      sizes="192x192"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-384x384.png"
      rel="apple-touch-icon"
      sizes="384x384"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-512x512.png"
      rel="apple-touch-icon"
      sizes="512x512"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-72x72.png"
      rel="icon"
      sizes="72x72"
      type="image/png"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-96x96.png"
      rel="icon"
      sizes="96x96"
      type="image/png"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-192x192.png"
      rel="icon"
      sizes="192x192"
      type="image/png"
    />
    <link
      href="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-512x512.png"
      rel="apple-touch-startup-image"
    />
    <link
      href="/favicon.ico"
      rel="shortcut icon"
      type="image/x-icon"
    />
    <meta
      content="IE=edge"
      http-equiv="X-UA-Compatible"
    />
    <meta
      charset="utf-8"
    />
    <meta
      content="noodp,noydir"
      name="robots"
    />
    <meta
      content="#B80000"
      name="theme-color"
    />
    <meta
      content="width=device-width, initial-scale=1, minimum-scale=1"
      name="viewport"
    />
    <meta
      content="BBC News አማርኛ"
      name="apple-mobile-web-app-title"
    />
    <meta
      content="BBC News አማርኛ"
      name="application-name"
    />
    <meta
      content="ዝግጅቶቻችንን’"
      name="description"
    />
    <meta
      content="100004154058350"
      name="fb:admins"
    />
    <meta
      content="1609039196070050"
      name="fb:app_id"
    />
    <meta
      content="yes"
      name="mobile-web-app-capable"
    />
    <meta
      content="#B80000"
      name="msapplication-TileColor"
    />
    <meta
      content="https://news.files.bbci.co.uk/include/articles/public/amharic/images/icons/icon-144x144.png"
      name="msapplication-TileImage"
    />
    <meta
      content="ዝግጅቶቻችንን’"
      name="og:description"
    />
    <meta
      content="https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png"
      name="og:image"
    />
    <meta
      content="BBC News አማርኛ"
      name="og:image:alt"
    />
    <meta
      content="am-ET"
      name="og:locale"
    />
    <meta
      content="BBC News አማርኛ"
      name="og:site_name"
    />
    <meta
      content="ያድምጡ - BBC News አማርኛ"
      name="og:title"
    />
    <meta
      content="website"
      name="og:type"
    />
    <meta
      content="https://www.test.bbc.com/pathname"
      name="og:url"
    />
    <meta
      content="summary_large_image"
      name="twitter:card"
    />
    <meta
      content="@bbcnews"
      name="twitter:creator"
    />
    <meta
      content="ዝግጅቶቻችንን’"
      name="twitter:description"
    />
    <meta
      content="BBC News አማርኛ"
      name="twitter:image:alt"
    />
    <meta
      content="https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png"
      name="twitter:image:src"
    />
    <meta
      content="@bbcnews"
      name="twitter:site"
    />
    <meta
      content="ያድምጡ - BBC News አማርኛ"
      name="twitter:title"
    />
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"RadioChannel","url":"https://www.test.bbc.com/pathname","publisher":{"@type":"NewsMediaOrganization","name":"BBC News አማርኛ","publishingPrinciples":"https://www.bbc.com/news/help-41670342","logo":{"@type":"ImageObject","width":1024,"height":576,"url":"https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png"}},"image":{"@type":"ImageObject","width":1024,"height":576,"url":"https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png"},"thumbnailUrl":"https://news.files.bbci.co.uk/ws/img/logos/og/amharic.png","mainEntityOfPage":{"@type":"WebPage","@id":"https://www.test.bbc.com/pathname","name":"ያድምጡ"}}
    </script>
  </head>
  <body>
    <div>
      <noscript />
      <main
        class="c0 c1"
        dir="ltr"
        role="main"
      >
        <div
          class="c2"
          dir="ltr"
        >
          <h1
            class="c3"
            id="content"
            tabindex="-1"
          >
            ያድምጡ
          </h1>
          <p
            class="c4"
          >
            ዝግጅቶቻችንን’
          </p>
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <iframe
                  allow="autoplay; fullscreen"
                  allowfullscreen=""
                  class="c8"
                  scrolling="no"
                  src="https://polling.test.bbc.co.uk/ws/av-embeds/media/bbc_amharic_radio/liveradio/am"
                  title="Audio player"
                />
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>
</html>
`;
